src/main/java/bio/terra/common/ErrorCollector.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.common;
+
+import bio.terra.service.dataset.exception.IngestFailureException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ErrorCollector {
+  private List<String> errors;
+  private int maxBadLoadFileLineErrorsReported;
+  private String exceptionMessage;
+
+  public ErrorCollector(int maxBadLoadFileLineErrorsReported, String exceptionMessage) {
+    this.errors = new ArrayList<>();
+    this.maxBadLoadFileLineErrorsReported = maxBadLoadFileLineErrorsReported;
+    this.exceptionMessage = exceptionMessage;
+  }
+
+  public boolean anyErrorsCollected() {
+    return errors.size() > 0;
+  }
+
+  public void record(String lineErrorMsgFormat, String lineErrorMsg) {
+    if (errors.size() < maxBadLoadFileLineErrorsReported) {
```

Comment:
It shouldn't - It will record the error until the max is hit. Once the max is hit (and therefore the else statement is called), an exception is thrown and we don't continue.

Reply:
That's true, but it makes this code dependent on the behavior of the code that uses it. If the caller decides to catch the exception and then call `record()` again, they'll end up with another truncate message. In the scheme of things this is a really minor point and the code works fine either way (and using the error collector in this way is probably "wrong"). But I think it's better to make the code work regardless of how it's used, if it doesn't add too much complexity.

Reply:
Fair enough - I've switched the 'else' statement to `else if (errors.size() == maxErrorsReported)` and now it doesn't do anything if we've already hit the max.